metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/dereference/DereferencerImpl.java

```diff
@@ -33,315 +37,321 @@
-    if (CollectionUtils.isNotEmpty(remainingReferences)) {
-      DereferencedEntities aggregationRemainingDereferencingResult = dereferenceExternalEntity(remainingReferences, classType);
-      result.getReferenceTermListMap().putAll(aggregationRemainingDereferencingResult.getReferenceTermListMap());
-      result.getReportMessages().addAll(aggregationRemainingDereferencingResult.getReportMessages());
-    }
+        //Collect references that returned empty lists values for references that we checked with uri
+        Set<ReferenceTerm> remainingReferences = referenceTerms.stream().filter(
+                        referenceTerm -> result.getReferenceTermListMap().get(referenceTerm).isEmpty())
+                .collect(Collectors.toSet());
 
-    return result;
-  }
+        //If there are any remaining references then do external dereferencing
+        if (CollectionUtils.isNotEmpty(remainingReferences)) {
+            DereferencedEntities aggregationRemainingDereferencingResult = dereferenceExternalEntity(remainingReferences, classType);
+            result.getReferenceTermListMap().putAll(aggregationRemainingDereferencingResult.getReferenceTermListMap());
+            result.getReportMessages().addAll(aggregationRemainingDereferencingResult.getReportMessages());
+        }
 
-  @Override
-  public Map<Class<? extends AboutType>, Set<String>> extractReferencesForDereferencing(RDF rdf) {
-    return DereferenceUtils.extractReferencesForDereferencing(rdf);
-  }
+        return result;
+    }
 
-  private DereferencedEntities dereferenceOwnEntities(Set<ReferenceTerm> resourceIds,
-      HashSet<Report> reports,
-      Class<? extends AboutType> classType) {
-    if (entityResolver == null) {
-      return new DereferencedEntities(Collections.emptyMap(), new HashSet<>());
+    @Override
+    public Map<Class<? extends AboutType>, Set<String>> extractReferencesForDereferencing(RDF rdf) {
+        return DereferenceUtils.extractReferencesForDereferencing(rdf);
     }
-    try {
-      Map<ReferenceTerm, List<EnrichmentBase>> result = new HashMap<>();
-      entityResolver.resolveById(resourceIds).forEach((key, value) -> result.put(key, List.of(value)));
-      return new DereferencedEntities(result, reports, classType);
-    } catch (Exception e) {
-      return handleDereferencingException(resourceIds, reports, e, classType);
+
+    private DereferencedEntities dereferenceOwnEntities(Set<ReferenceTerm> resourceIds,
+                                                        HashSet<Report> reports,
+                                                        Class<? extends AboutType> classType) {
+        if (entityResolver == null) {
+            return new DereferencedEntities(Collections.emptyMap(), new HashSet<>());
+        }
+        try {
+            Map<ReferenceTerm, List<EnrichmentBase>> result = new HashMap<>();
+            entityResolver.resolveById(resourceIds).forEach((key, value) -> result.put(key, List.of(value)));
```

Comment:
Here you probably need to check against the europeana URL. 
Something like
```
var ownEntities = resourceIds.stream()
    .filter(id->EntityResolver.europeanaLinkPattern.matcher(id.getReference().toString()).matches())
    .collect(Collectors.toSet());
entityResolver.resolveById(ownEntities).forEach((key, value) -> result.put(key, List.of(value)));
ownEntities.stream().filter(id->!result.containsKey(id))
    .forEach(notFoundOwnId ->setDereferenceStatusInReport(notFoundOwnId.getReference().toString(), 
        reports, DereferenceResultStatus.UNKNOWN_EUROPEANA_ENTITY));
```

Reply:
I understand what this means now, because of the previous way the status was computed never thought to explicitly validate it. will proceed to update the code.